pystac/item.py

```diff
@@ -22,11 +22,11 @@ class Item(STACObject):
         geometry (dict): Defines the full footprint of the asset represented by this item,
             formatted according to `RFC 7946, section 3.1 (GeoJSON)
             <https://tools.ietf.org/html/rfc7946>`_.
-        bbox (List[float]):  Bounding Box of the asset represented by this item using
-            either 2D or 3D geometries. The length of the array must be 2*n where n is the
-            number of dimensions.
         datetime (datetime): Datetime associated with this item.
         properties (dict): A dictionary of additional metadata for the item.
+        bbox (List[float] or None):  Bounding Box of the asset represented by this item using
```

Comment:
Merging, but making a note of how I will change this - I'm going to revert back to the bbox's original position and have it not default to None. This is for a couple of reasons:
- This change will break the API and make upgrades more difficult, which isn't too heavy of a price but always good to avoid where possible
- The null geometry case is special, and so having a default of None goes against the standard default case (where bbox is required).

I think having the user be explicit about passing in None for the special null-geometry case is fine, and better than potential confusion about whether or not the bbox is required in the standard case where there is a geometry.